conda/cli/main_rename.py

```diff
@@ -0,0 +1,66 @@
+# -*- coding: utf-8 -*-
+# Copyright (C) 2012 Anaconda, Inc
+# SPDX-License-Identifier: BSD-3-Clause
+from __future__ import annotations
+
+from functools import partial
+import os
+
+from conda.base.context import context, locate_prefix_by_name, validate_prefix_name
+from conda.base.constants import DRY_RUN_PREFIX
+from conda.cli import common, install
+from conda.common.path import expand
+from conda.exceptions import CondaEnvException
+from conda.gateways.disk.delete import rm_rf
+
+
+def validate_src(args) -> str:
+    """
+    Validate that the "base" environment is not being renamed and that the
+    currently active prefix isn't trying to be renamed.
+    """
+    if context.target_prefix == context.root_prefix:
+        raise CondaEnvException("The 'base' environment cannot be renamed")
+
+    prefix = args.name if args.name else args.prefix
+
+    if common.is_active_prefix(prefix):
+        raise CondaEnvException("Cannot rename the active environment")
+
+    return locate_prefix_by_name(prefix)
+
+
+def validate_destination(dest: str, force: bool = False) -> str:
+    """Ensure that our destination does not exist"""
+    if os.sep in dest:
+        dest = expand(dest)
+    else:
+        dest = validate_prefix_name(dest, ctx=context, allow_base=False)
+
+    if not force and os.path.exists(dest):
+        raise CondaEnvException("Environment destination already exists. Override with --force.")
+
+    return dest
+
+
+def execute(args, _):
+    """
+    Executes the command for renaming an existing environment
+    """
+    src = validate_src(args)
+    dest = validate_destination(args.destination, force=args.force)
+
+    actions: list[partial] = []
+
+    if args.force:
+        actions.append(partial(rm_rf, dest))
```

Comment:
Should this be atomic? What if the rename ends up failing? The target environment gets deleted "for nothing". Maybe rename to something else first, and if everything succeeded, end up deleting for good. If something failed, undo the initial rename.

Reply:
@jaimergp, great recommendation! I just implemented this in a recent commit. Let me know what you think.